src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,225 @@
+     */
+    public Parser(String filepath) {
+        this.ui = new Ui();
+        this.fileStorage = new FileStorage(filepath);
+        this.fullList = new TaskList();
+    }
+
+    /**
+     * Constructs a Parser instance with the specified parameters.
+     *
+     * @param fileStorage The FileStorage object to handle file operations.
+     * @param fullList The TaskList object to store ttasks.
+     * @param ui The Ui object to display outputs to the user.
+     */
+    public Parser(FileStorage fileStorage, TaskList fullList, Ui ui) {
+        this.ui = ui;
+        this.fileStorage = fileStorage;
+        this.fullList = fullList;
+    }
+
+    /**
+     * Represents the various command types that can be used in the chatBot application.
+     * These command types define the actions the chatBot can perform.
+     */
+    public enum CommandType {
+        TODO, DEADLINE, EVENT, MARK, UNMARK,
+        DELETE, LIST, BYE, UNKNOWN, EMPTY, FIND
+    }
+
+    /**
+     * Returns a CommandType based on the use input.
+     *
+     * @param input User input to be parsed.
+     * @return The CommandType enum representing the parsed user input.
+     */
+    public CommandType parseCommand(String input) {
+        if (input.equals("bye")) {
+            return CommandType.BYE;
+        } else if (input.equals("mark")) {
+            return CommandType.MARK;
+        } else if (input.equals("unmark")) {
+            return CommandType.UNMARK;
+        } else if (input.equals("todo")) {
+            return CommandType.TODO;
+        } else if (input.equals("deadline")) {
+            return CommandType.DEADLINE;
+        } else if (input.equals("event")) {
+            return CommandType.EVENT;
+        } else if (input.equals("delete")) {
+            return CommandType.DELETE;
+        } else if (input.equals("list")) {
+            return CommandType.LIST;
+        } else if (input.equals("")) {
+            return CommandType.EMPTY;
+        } else if (input.equals("find")) {
+            return CommandType.FIND;
+        } else {
+            return CommandType.UNKNOWN;
+        }
+    }
+
+    /**
+     * Converts a date string to a LocalDateTime object.
+     *
+     * @param date The date string in the specified format.
+     * @return A LocalDateTime object representing the parsed date and time.
+     */
+    public LocalDateTime setDate(String date) {
+        try {
+            return LocalDateTime.parse(date, formatter);
+        } catch (DateTimeParseException e) {
+            System.out.println("Invalid date input. Reverting to current date and time");
+            return LocalDateTime.now();
+        }
+    }
+
+    /**
+     * Processes user input to create and execute task-related actions.
+     *
+     * @param input The user input to be processed.
+     * @return {@code true} if the application should continue running. {@code false} if it should exit.
+     */
+    public boolean createTaskAction(String input) {
```

Comment:
I like the rationale behind this method! I think a more intuitive name for this method could be `isTaskCreated`, so that when false is returned, it means that the task was not created, and the application should stop running. What do you think?